app/controllers/coupons_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CouponsController < ApplicationController
+  def index
+    @coupons = Coupon.all
+  end
+
+  def show
+    @coupon = Coupon.find(params[:id])
+  end
+
+  def new
+    @coupon = Coupon.new
+  end
```

Comment:
Since you're not using `@coupon` in the view, you don't need this. 
```suggestion
  def new; end
```